src/prefect/telemetry/services.py

```diff
@@ -0,0 +1,67 @@
+from abc import abstractmethod
+from typing import Union
+
+from opentelemetry.exporter.otlp.proto.http._log_exporter import OTLPLogExporter
+from opentelemetry.exporter.otlp.proto.http.trace_exporter import OTLPSpanExporter
+from opentelemetry.sdk._logs import LogData
+from opentelemetry.sdk._logs.export import LogExporter
+from opentelemetry.sdk.trace import ReadableSpan
+from opentelemetry.sdk.trace.export import SpanExporter
+
+from prefect._internal.concurrency.services import BatchedQueueService
+
+
+class BaseQueueingExporter(BatchedQueueService):
+    _max_batch_size = 512
+    _min_interval = 5.0
+    _otlp_exporter: Union[SpanExporter, LogExporter]
+
+    def export(self, batch: list) -> None:
+        for item in batch:
+            self.send(item)
+
+    @abstractmethod
+    def _export_batch(self, items: list) -> None:
+        pass
+
+    async def _handle_batch(self, items: list) -> None:
```

Comment:
are these not getting `list[T]` because of the base class? if so, makes total sense we wouldnt update it here 👍

Reply:
Yeah, I could make it `list[Union[...]]`, let me see if mypy likes that.